### PR TITLE
Container type: document that SecurityContext applies to Container

### LINKED ISSUE #90634: Container API: document that SecurityContext applies to the Container instead of Pod
**What happened**:
types.go states that Container.SecurityContext applies to a Pod, which can be misleading since we already have PodSecurityContext.

`Security options the pod should run with.`

**What you expected to happen**:
Documentation states that it applies to container

**How to reproduce it (as minimally and precisely as possible)**:
-

**Anything else we need to know?**:
-

**Environment**:
-

### PR DESCRIPTION
**Pull request by desaintmartin**
Container type: document that SecurityContext applies to Container instead of Pod.

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
/kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

types.go states that Container.SecurityContext applies to a Pod, which can be misleading since we already have PodSecurityContext.

**Which issue(s) this PR fixes**:

Fixes #90634

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/api/core/v1/types.go [hunk 1]
```diff
@@ -2245,7 +2245,7 @@ type Container struct {
 	// More info: https://kubernetes.io/docs/concepts/containers/images#updating-images
 	// +optional
 	ImagePullPolicy PullPolicy `json:"imagePullPolicy,omitempty" protobuf:"bytes,14,opt,name=imagePullPolicy,casttype=PullPolicy"`
-	// Security options the pod should run with.
+	// Security options the container should run with.
```
Comment (mikedanese) [inline staging/src/k8s.io/api/core/v1/types.go:2248]: > SecurityContext defines the security options the container should be run with. If set, the fields of SecurityContext override the equivalent fields of PodSecurityContext.

To match internal API doc?

Comment (desaintmartin) [reply staging/src/k8s.io/api/core/v1/types.go]: Done!

**Comment by desaintmartin** [comment]
Gentle up.

**Comment by liggitt** [comment]
thanks. you'll need to run `make update` to propagate the doc change into the various generated files.

**Comment by desaintmartin** [comment]
Done!

**Comment by desaintmartin** [comment]
Should I update it or close it?

**Review by sftim — COMMENTED** [review]
/ok-to-test

**Comment by liggitt** [comment]
/lgtm
/approve

this lgtm, but will conflict with https://github.com/kubernetes/kubernetes/pull/98740

suggest rebasing once that merges

**Comment by liggitt** [comment]
/hold

actually, can you squash the commits?

**Comment by desaintmartin** [comment]
Rebased & squashed.
/retest

**Pull request merged**